packages/swing-store/src/swingStore.js

```diff
@@ -525,6 +541,37 @@ function makeSwingStore(dirPath, forceReset, options) {
     stopTrace();
   }
 
+  /**
+   * Return a Buffer with the entire DB state, useful for cloning a
+   * small swingstore in unit tests.
+   *
+   * @returns {Buffer}
+   */
+  function serialize() {
+    // An on-disk DB with WAL mode enabled seems to produce a
+    // serialized Buffer that can be unserialized, but the resulting
+    // 'db' object fails all operations with SQLITE_CANTOPEN. So
+    // pre-emptively throw.
+    if (filePath !== ':memory:') {
+      throw Error('on-disk DBs with WAL mode enabled do not serialize well');
+    }
+    return db.serialize();
+  }
+
+  function getAllKVEntries() {
```

Comment:
For symmetry, I'm wondering if `getAllKVEntries` should be called something like `dumpKVEntries`.

I'm also wondering if at this point we should split all the kvStore stuff into its own source file, parallel to `snapStore.js` and `streamStore.js`.  (Not in this PR, but if you agree it makes sense then one of us should add an issue.)

Reply:
yes on both, I'll file an issue and I can do it after we land the `getNext` stuff (just to minimize merge conflicts)

Reply:
actually I'll do the `dumpKVEntries` here, it's trivial, but I'll defer the refactoring for later, the activityhash made that nontrivial